mods/taskbar-music-lounge.wh.cpp

```diff
@@ -417,8 +503,28 @@ void DrawMediaPanel(HDC hdc, int width, int height) {
 LRESULT CALLBACK MediaWndProc(HWND hwnd, UINT msg, WPARAM wParam, LPARAM lParam) {
     switch (msg) {
         case WM_CREATE: 
-            UpdateAppearance(hwnd); // Apply DWM Rounding + Acrylic
+            UpdateAppearance(hwnd); 
             SetTimer(hwnd, IDT_POLL_MEDIA, 1000, NULL); 
+            
+            // Register Event Hook scoped to Taskbar Thread
+            {
+                HWND hTaskbar = FindWindow(L"Shell_TrayWnd", nullptr);
+                if (hTaskbar) {
+                    DWORD pid = 0;
+                    DWORD tid = GetWindowThreadProcessId(hTaskbar, &pid);
+                    if (tid != 0) {
+                        g_TaskbarHook = SetWinEventHook(

```

Comment:
What happens if explorer is restarted? You might want to handle it (and make sure other things work) by handling the "TaskbarCreated" message:
https://learn.microsoft.com/en-us/windows/win32/shell/taskbar#taskbar-creation-notification